src/main/java/core/basesyntax/strategy/FruitTransaction.java

```diff
@@ -0,0 +1,51 @@
+package core.basesyntax.strategy;
+
+public class FruitTransaction {
+    private final Operation operation;
+    private final String fruit;
+    private final int quantity;
+
+    public FruitTransaction(Operation operation, String fruit, int quantity) {
+        this.operation = operation;
+        this.fruit = fruit;
+        this.quantity = quantity;
+    }
+
+    public Operation getOperation() {
+        return operation;
+    }
+
+    public String getFruit() {
+        return fruit;
+    }
+
+    public int getQuantity() {
+        return quantity;
+    }
+
+    public enum Operation {
+        BALANCE("b"),
+        SUPPLY("s"),
+        PURCHASE("p"),
+        RETURN("r");
+
+        private final String code;
+
+        Operation(String code) {
+            this.code = code;
+        }
+
+        public static Operation fromCode(String code) {
+            for (Operation o : values()) {
+                if (code == null) {
+                    throw new NullPointerException();
+                }
+                if (o.code.equals(code)) {
+                    return o;
+                }
+            }
+
+            throw new IllegalArgumentException("Unknown operation code: " + code);
```

Comment:
```suggestion
            for (Operation operation : values()) {
                if (operation.code.equals(code)) {
                    return operation;
                }
            }
            throw new IllegalArgumentException("Unknown operation code: " + code);
```

Reply:
I get it. So there is no need to check for a null argument in the tests? Thanx.